test/Test/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -187,6 +187,8 @@ tests = testGroup "Database.LSMTree.Internal.RawPage"
     , testProperty "entry" prop_single_entry
     , testProperty "rawPageOverflowPages" prop_rawPageOverflowPages
     , testProperty "from/to reference impl" prop_fromToReferenceImpl
+    , testProperty "∀ key page. maybe True (key <=) (getRawPageIndexKey . rawPageIndex page .  id  =<< rawPageFindKey page key)" prop_findKey_index_law_GT
+    , testProperty "∀ key page. maybe True (key > ) (getRawPageIndexKey . rawPageIndex page . pred =<< rawPageFindKey page key)" prop_findKey_index_law_LT
```

Comment:
These tests only assert anything when `rawPageFindKey` returns `Just`. Can you also add a property that whenever the last key of the page is `(>= key)`, then we successfully find an index number?

Looking at the code I have a hunch that it might sometimes incorrectly return `Nothing`, but I could be wrong.

Reply:
@mheinzel Your hunch seems to have been accurate. I added a property test for the `Nothing` case and it does fail! Any thoughts on when it incorrectly returns `Nothing`?

Reply:
You can replicate a failure case on this branch by invoking:

```
cabal test lsm-tree-test --test-options='--quickcheck-replay="(SMGen 3937336630516587970 3943216271980923571,0)" -p "/rawPageFindKey missing/"'
```

Reply:
@mheinzel I have figured out the corner case in the binary search!

Reply:
Nice!

The properties we seem a little complex by now. On the other hand, it is quite straightforward to to create a reference implementation working on the list of k/ops, similar to the test for `rawPageLookup`. Maybe that would be a nicer approach?